realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1472,6 +1472,72 @@ public long count() {
         return this.query.count();
     }
 
+    RealmResults<E> distinctAsync(final long columnIndex) {
```

Comment:
I would prefer that `distinctAsync()` is a method in `Realm` and `io.realm.internal.Table`. When a user reads the API doc, it is a bit confusing that `realm.where(Person.class).equalTo("name", "John").distinctAsync(5)` is not allowed (5 = field/column number for the `age` field).